### PR TITLE
Support version tags that don't start with v for semantic versioning

### LINKED ISSUE #242: hasAnyVersionTags not counting semver tags that doesn't start with 'v'
**Bug description**

**How to reproduce**

Create tags that doesn't start with `v`. i.e. `0.0.1`, `1.0.0`, `1.0`, etc.

```bash
git tag --list
```

```
1
1.2
1.2.3
11.11.11
2,2,2
v0HELLO
v1
v1.2
v1.2.3
v11.11.11
x1.0.0
```

[Current](https://github.com/game-ci/unity-builder/blob/398eda622fffdac0b8e33e01f21c4c47fe24a955/src/model/versioning.ts#L270)
```bash
git tag --list | grep v[0-9]*
```
```
v0HELLO
v1
v1.2
v1.2.3
v11.11.11
```

**Expected behavior**

```bash
git tag --list | grep '^v\?[0-9]\+\(\.[0-9]\+\)\?\(\.[0-9]\+\)\?$'
```
```
1
1.2
1.2.3
11.11.11
v1
v1.2
v1.2.3
v11.11.11
```

```diff
-      input: Buffer.from('git tag --list --merged HEAD | grep v[0-9]* | wc -l'),
+      input: Buffer.from('git tag --list --merged HEAD | grep '^v\?[0-9]\+\(\.[0-9]\+\)\?\(\.[0-9]\+\)\?$' | wc -l'),
```

Or more permissive:

```bash
git tag --list | grep '^v\?[0-9]\+'
```
```
1
1.2
1.2.3
11.11.11
2,2,2
v0HELLO
v1
v1.2
v1.2.3
v11.11.11
```

```diff
-      input: Buffer.from('git tag --list --merged HEAD | grep v[0-9]* | wc -l'),
+      input: Buffer.from('git tag --list --merged HEAD | grep '^v\?[0-9]\+' | wc -l'),
```

### PR DESCRIPTION
**Pull request by davidmfinol**
#### Changes

Fixes #242

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Review by GabLeRoux — APPROVED** [review]

**Comment by GabLeRoux** [comment]
I totally agree with this. According to https://semver.org/ prefixing with `v` isn't considered semantic so allowing versions without it should definitely be considered 😅 

> Is “v1.2.3” a semantic version?
>
> No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

**Comment by webbertakken** [comment]
> I totally agree with this. According to https://semver.org/ prefixing with `v` isn't considered semantic so allowing versions without it should definitely be considered 😅
> 
> > Is “v1.2.3” a semantic version?
> > No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

While I agree we should support this, the `v` is not specific to `semantic versioning` and not part of the semantic version itself. It is GitHub convention to use `v` in the tag to indicate it's a version tag.

File: src/model/versioning.test.ts [hunk 1]
```diff
@@ -125,6 +125,7 @@ describe('Versioning', () => {
       'is happy with valid %s',
       (description) => {
         expect(Versioning.descriptionRegex1.test(description)).toBeTruthy();
+        expect(Versioning.descriptionRegex1.test(description?.slice(1))).toBeTruthy();
```
Comment (webbertakken) [inline src/model/versioning.test.ts:128]: Suggest adding a separate tests for `it('accepts valid semantic versions without v-prefix', () => {})`

**Review by webbertakken — APPROVED** [review]

**Comment by webbertakken** [comment]
Two more things:

- Would you mind updating semver-regex as part of this PR?
image
- Could you please use descriptive PR titles, so that the releases properly show what changed (without manual work). See description generator: https://takken.io/tools

File: src/model/versioning.test.ts (hunk 1 above)
Comment (davidmfinol) [reply src/model/versioning.test.ts]: Done

**Comment by davidmfinol** [comment]
> * Would you mind updating semver-regex as part of this PR?
I didn't want to manually update the yarn.lock file, so I found that the semver-regex dependency came from husky, and I updated the versions of husky and lint-staged. The semver-regex dependency is now gone.

> * Could you please use descriptive PR titles, so that the releases properly show what changed (without manual work). See description generator: https://takken.io/tools
I was hoping that the name would come from the linked issue, but I guess it doesn't do that. I've updated the title.

File: .husky/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+_
```
Comment (webbertakken) [inline .husky/.gitignore:1]: what does this do exactly?

**Review by webbertakken — APPROVED** [review]

File: .husky/.gitignore (hunk 2 above)
Comment (davidmfinol) [reply .husky/.gitignore:1]: Yeah, it's auto-generated. The `husky/_` folder seems to be generated during the pre-commit check as a temporary workspace

**Pull request merged**